ui/packages/app/src/project_setting/metadata/MetadataList.js

```diff
@@ -0,0 +1,190 @@
+
+const MetadataList = ({ project, fetchUpdates }) => {
+  const streamOptions = useMemo(() => {
+    return Object.entries(config.STREAMS)
+      .map(([stream]) => stream.trim())
+      .sort((a, b) => a.localeCompare(b))
+      .map(stream => ({ label: stream }));
+  }, []);
+
+  const [selectedStream, setStream] = useState(project.stream);
+  const [streamError, setStreamError] = useState("");
+  const [isValidStream, setIsValidStream] = useState(false);
+  const onStreamChange = selectedStream => {
+    let isValid = isDNS1123Label(selectedStream);
+    if (!isValid) {
+      setStreamError(
+        "Stream name is invalid. It should contain only lowercase alphanumeric and dash (-)"
+      );
+    }
+    setIsValidStream(isValid);
+    setStream(selectedStream);
+    setTeam(undefined);
+  };
+
+  const teamOptions = useMemo(() => {
+    return (config.STREAMS[selectedStream] || [])
+      .sort((a, b) => a.localeCompare(b))
+      .map(team => ({ label: team.trim() }));
+  }, [selectedStream]);
+
+  const [selectedTeam, setTeam] = useState(project.team);
+  const [teamError, setTeamError] = useState("");
+  const [isValidTeam, setIsValidTeam] = useState(false);
+  const onTeamChange = selectedTeam => {
+    if (selectedTeam !== project.team) {
+      let isValid = isDNS1123Label(selectedTeam);
+      if (!isValid) {
+        setTeamError(
+          "Team name is invalid. It should contain only lowercase alphanumeric and dash (-)"
+        );
+      }
+      setIsValidTeam(isValid);
+      setTeam(selectedTeam);
+    }
+    if (selectedTeam === project.team && selectedStream === project.stream) {
+      setIsValidTeam(false);
+      setIsValidStream(true);
+      setTeam(selectedTeam);
+    } else if (
+      selectedTeam !== project.team &&
+      selectedStream === project.stream
+    ) {
+      setIsValidTeam(true);
+      setIsValidStream(true);
+      setTeam(selectedTeam);
+    }
+  };
+
+  useEffect(() => {
+    if (!selectedTeam) {
+      setIsValidTeam(false);
+    }
+  }, [selectedTeam]);
+
+  const [selectedLabels, setLabels] = useState(project.labels);
+  const [isValidLabels, setIsValidLabels] = useState(true);
+  const [labelError, setLabelError] = useState("");
+  const onLabelChange = labels => {
+    const labelsValid =
+      labels.length === 0
+        ? true
+        : labels.reduce((labelsValid, label) => {
+            return labelsValid && label.isKeyValid && label.isValueValid;
+          }, true);
+    setIsValidLabels(labelsValid);
+    if (!labelsValid) {
+      setLabelError(
+        "Invalid labels. Both key and value of a label must contain only alphanumeric and dash (-)"
+      );
+    }
+
+    if (
+      selectedTeam === project.team &&
+      selectedStream === project.stream &&
+      selectedLabels !== project.labels
+    ) {
+      setIsValidTeam(true);
+      setIsValidStream(true);
+      setIsValidLabels(labelsValid);
+      setTeam(selectedTeam);
+      setStream(selectedStream);
+    }
```

Comment:
Same comment as the other one on this file. Maybe it's really not necessary to check if any of the fields have changed to determine whether or not the submit button should be disabled.